packages/bull-vault/test/e2e/BullStrategyTestFork.t.sol

```diff
@@ -1,43 +1,179 @@
 
-//         // to execute tx on behalf user1 
-//         // vm.prank(user1);
-//         squeeth.approve(address(bullStrategy), type(uint256).max);
-//     }
-//     function testDeposit() public {}
+        user1Pk = 0xA11CE;
+        user1 = vm.addr(user1Pk);
 
-// }
\ No newline at end of file
+        vm.label(user1, "User 1");
+        vm.label(address(bullStrategy), "BullStrategy");
+        vm.label(euler, "Euler");
+        vm.label(eulerMarketsModule, "EulerMarkets");
+        vm.label(usdc, "USDC");
+        vm.label(weth, "WETH");
+
+        vm.deal(user1, 100000000e18);
+        // this is a crab whale, get some crab token from
+        vm.prank(0x06CECFbac34101aE41C88EbC2450f8602b3d164b);
+        IERC20(crabV2).transfer(user1, 100e18);
+        // some WETH and USDC rich address
+        vm.prank(0x57757E3D981446D585Af0D9Ae4d7DF6D64647806);
+        IERC20(weth).transfer(user1, 10000e18);
+    }
+
+    function testInitialDeposit() public {
+        uint256 crabToDeposit = 10e18;
+        (uint256 wethToLend, uint256 usdcToBorrow) = _calcCollateralAndBorrowAmount(crabToDeposit);
+
+        vm.startPrank(user1);
+        IERC20(crabV2).approve(address(bullStrategy), crabToDeposit);
+        bullStrategy.deposit{value: wethToLend}(crabToDeposit);
+        vm.stopPrank();
+
+        assertEq(bullStrategy.balanceOf(user1), crabToDeposit);
+        assertEq(IEulerDToken(dToken).balanceOf(address(bullStrategy)), usdcToBorrow);
+        assertTrue(wethToLend.sub(IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy))) <= 1);
+        assertEq(IERC20(usdc).balanceOf(user1), usdcToBorrow);
+    }
+
+    // fuzz testing is awesome!
+    function testFuzzingDeposit(uint256 _crabAmount) public {
+        vm.assume(_crabAmount < IERC20(crabV2).balanceOf(user1));
+        vm.assume(_crabAmount > 0);
+
+        uint256 bullToMint = _calcBullToMint(_crabAmount);
+        (uint256 wethToLend, uint256 usdcToBorrow) = _calcCollateralAndBorrowAmount(_crabAmount);
+        uint256 userBullBalanceBefore = bullStrategy.balanceOf(user1);
+        uint256 ethInLendingBefore = IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy));
+        uint256 usdcBorrowedBefore = IEulerDToken(dToken).balanceOf(address(bullStrategy));
+        uint256 userUsdcBalanceBefore = IERC20(usdc).balanceOf(user1);
+
+        vm.startPrank(user1);
+        IERC20(crabV2).approve(address(bullStrategy), _crabAmount);
+        bullStrategy.deposit{value: wethToLend}(_crabAmount);
+        vm.stopPrank();
+        
+        assertEq(bullStrategy.balanceOf(user1).sub(userBullBalanceBefore), bullToMint);
+        assertTrue(wethToLend.sub(IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy)).sub(ethInLendingBefore)) <= 2);
+        assertEq(IEulerDToken(dToken).balanceOf(address(bullStrategy)).sub(usdcBorrowedBefore), usdcToBorrow);
+        assertEq(IERC20(usdc).balanceOf(user1).sub(userUsdcBalanceBefore), usdcToBorrow);
```

Comment:
should probably have some more assertions - crab received is what you expect, crab withdrawn is what you expect